kalliope/tts/googletts/googletts.py

```diff
@@ -43,35 +46,83 @@ def _generate_audio_file(self):
-        # Verify the response status code and the response content type
-        if r.status_code != requests.codes.ok or content_type != TTS_CONTENT_TYPE:
-            raise FailToLoadSoundFile("Googletts : Fail while trying to remotely access the audio file")
 
-        # OK we get the audio we can write the sound file
-        FileManager.write_in_file(self.file_path, r.content)
+            # Verify the response status code and the response content type
+            if r.status_code != requests.codes.ok or content_type != TTS_CONTENT_TYPE:
+                raise FailToLoadSoundFile("Googletts : Fail while trying to remotely access the audio file")
+
+            # OK we get the audio we can write the sound file
+            FileManager.write_in_file(self.file_path + str(i), r.content)
+            i += 1
+
+        # Stitch together all of the responses
+        master_file = AudioSegment.from_mp3(self.file_path + "0")
+        for x in range(1, i):
+            master_file += AudioSegment.from_mp3(self.file_path + str(x))
 
-    def get_payload(self):
+        master_file.export(self.file_path, format="mp3")
+
+    def get_payloads(self):
         """
         Generic method used load the payload used to access the remote api
 
         :return: Payload to use to access the remote api
         """
-
-        return {
-            "q": self.words,
-            "tl": self.language,
-            "ie": "UTF-8",
-            "total": "1",
-            "client": "tw-ob"
-        }
+        qs = []
+        logger.debug("Googletts : %s", self.words)
+        if (self.split_sentences):
+            logger.debug("Googletts : splitting sentences")
+            request_text = ""
+            for c in self.words:
+                # If there is content in the request and we get to either the max or a delimiter, save the request
+                if (len(request_text) > 0 and
+                    c in SENTENCE_SPLIT_DELIMITERS or (len(request_text) >= MAX_REQUEST_LEN and c == ' ')):
+                    qs.append({
+                        "q": request_text + c,
+                        "tl": self.language,
+                        "ie": "UTF-8",
+                        "total": "1",
+                        "client": "tw-ob"
+                        })
+                    request_text = ""
+                elif (c not in SENTENCE_SPLIT_DELIMITERS):
+                    request_text += c
+
+            # get the remainder if any
+            if (len(request_text) > 0):
+                qs.append({
+                    "q": request_text,
+                    "tl": self.language,
+                    "ie": "UTF-8",
+                    "total": "1",
+                    "client": "tw-ob"
+                    })
+        else:
+            logger.debug("Googletts : request may be too large for Google to handle, enable the split_sentences option if this becomes a problem.")
+            qs.append({
+                "q": self.words,
```

Comment:
```
qs.append({
     "q": request_text,
      "tl": self.language,
        "ie": "UTF-8",
        "total": "1",
      "client": "tw-ob"
  })
```

Duplicate code: suggestion calculate the q variable and build the qs only at the end of the method.

Reply:
Could we replace "qs" variable name by something more explicite?